[IR TX RX] Library updates - New Protocols

jimmys01 opened the pull request.

jimmys01 committed:
[IR TX RX] Library updates - New Protocols

jimmys01 committed:
[IR TX RX] Plugin Updates

TD-er commented on src/_P016_IR.ino:
```diff
@@ -14,23 +15,13 @@
-#include <ir_Toshiba.h>
-#include <ir_Vestel.h>
-#include <ir_Whirlpool.h>
+#ifdef P035_Standardized_AC_Commands //They use the same library, so enable it for both plugins if already enabled in the other.
```
Why have `P035_....` checks here in the P016 file?
If there is some relation between both, we should have these defines in the file where all build dependent defines are managed. ( `define_plugin_sets.h` )

jimmys01 committed:
[IR TX RX] Lib Updates

jimmys01 committed:
[IR TX RX] Code refactoring

Lib updates gave me some ideas

jimmys01 commented:
[firmware.zip](https://github.com/letscontrolit/ESPEasy/files/3123607/firmware.zip) normal_IR_ESP8266_4M 
If anyone wants to test it. I Dont have much time lately, the last commits are untested.

TD-er commented:
Last commit cannot be built:
```
In file included from /home/travis/build/letscontrolit/ESPEasy/src/ESPEasy.ino:86:0:
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino: In function 'boolean Plugin_035(byte, EventStruct*, String&)':
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino:293:70: error: 'IrType_orig' was not declared in this scope
addLog(LOG_LEVEL_INFO, String(F("IRTX: IR Code Sent: ")) + IrType_orig);
^
src/ESPEasy-Globals.h:1519:60: note: in definition of macro 'addLog'
#define addLog(L,S) if (loglevelActiveFor(L)) { addToLog(L,S); }
^
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino:296:66: error: 'IrType_orig' was not declared in this scope
printWebString += String(F("IRTX: IR Code Sent: "))+ IrType_orig + String(F("<BR>"));
^
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino: At global scope:
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino:306:3: error: expected unqualified-id before 'return'
return success;
^
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino:307:1: error: expected declaration before '}' token
} // Plugin_035 END
^
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino: In function 'boolean Plugin_035(byte, EventStruct*, String&)':
/home/travis/build/letscontrolit/ESPEasy/src/_P035_IRTX.ino:305:3: error: control reaches end of non-void function [-Werror=return-type]
} // SWITCH END
^
cc1plus: some warnings being treated as errors
*** [.pioenvs/normal_IR_ESP8266_1M/src/ESPEasy.ino.cpp.o] Error 1
```

jimmys01 committed:
[IR RX TX] More Refactoring

jimmys01 commented:
[firmware.zip](https://github.com/letscontrolit/ESPEasy/files/3123867/firmware.zip)

TD-er commented:
I guess this one can also be merged?

jimmys01 commented:
Not tested yet, so let it sit a little more until i find time to test it

jimmys01 commented:
crankyoldgit  is way to active for me to keep up with the IR library those days.

TD-er commented:
Well, you don't have to stay in sync with him. For now we're getting more behind, since this one hasn't been merged yet.

TD-er commented:
Did you test it?

jimmys01 committed:
[IR TX RX] Update IR Lib

jimmys01 committed:
[IR TX RX] Add newest IR protocols in ESPEasy

jimmys01 commented:
I was hoping someone would jump and test it.. I have tested it, it works fine. If this gets past the Travis its fine for merge

jimmys01 committed:
[IR TX] Arbitrary bits re-enabled

Added back the ability to have arbitrary bits to the IR send command. Reflected that also in the received signal decoding. Using 0 bits as an argument the ir send command will still use the default protocol bits
ref: https://www.letscontrolit.com/forum/viewtopic.php?f=4&t=6452&p=37373#p37373

jimmys01 committed:
Merge pull request #2 from letscontrolit/mega

test

jimmys01 committed:
Update Json to ver 6

jimmys01 committed:
Migrating Json  from version 5 to 6

jimmys01 committed:
removed unused var

jimmys01 committed:
Unused var?

jimmys01 committed:
Unused var?

jimmys01 committed:
Merge pull request #3 from jimmys01/TX-RX

Tx rx

jimmys01 committed:
Merge pull request #4 from jimmys01/mega

pull

TD-er commented:
And the same for this PR, it is has now become impossible to merge as exclusive feature.

jimmys01 commented:
I have not wrapped my head arround github branches as it seems... any Ideas how to revert it?

jimmys01 committed:
Revert "Merge pull request #4 from jimmys01/mega"

This reverts commit a18e15fb95c950776f962f379191536f13cedc75, reversing
changes made to 4493ae5109875b05d221459acf735e4a449caff3.

jimmys01 committed:
Revert "Merge pull request #3 from jimmys01/TX-RX"

This reverts commit 0fa7862c511de1cda5c7503f492837863974b609, reversing
changes made to 919a4b0b1bc11a4ed9911f890d542531683e0166.
